miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/BarcodableDto.java

```diff
@@ -0,0 +1,65 @@
+package uk.ac.bbsrc.tgac.miso.dto;
+
+import static uk.ac.bbsrc.tgac.miso.core.data.Barcodable.EntityType;
+
+public class BarcodableDto {
+  private Long id;
+  private EntityType entityType;
```

Comment:
You probably want to make this a string for the DTO.

Reply:
Sure, I can do that.  Does a String have an advantage over an Enum?  When the object is converted to a JSON object, the EntityType is converted to a string, but we might not want to rely on that conversion.

Reply:
Assuming that the URL-resolution logic is in `BarcodableSearchRestController` (see comments below), we will need to `switch` on `entityType` eventually.  If we were to convert this field to a String, the `switch` might be a little less elegant.  Perhaps we should leave it as an `Enum`?

Reply:
Can add a `toDto(Barcodable)` method in the REST controller that converts to DTO as normal with String property, and then adds the URL by looking at the enum value on the Barcodable. Map with that instead of the two current map operations

Reply:
:+1: